File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.c
int smallestIndex(int* nums, int numsSize) {
    int sum = 0;
    int result = -1;
    for(int i = 0; i < numsSize; i++){
        int curr_num = nums[i];
        while(curr_num != 0){
            int temp = curr_num % 10;
            sum += temp;
            curr_num /= 10;
        }

        if(sum == i){
            result = i;
            break;
        }else{
            sum = 0;
        }
    }
    return result;
}

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        total_sum = 0
        result = -1
        for i in range(len(nums)):
            curr_num = nums[i]
            while curr_num != 0:
                temp = curr_num % 10
                total_sum += temp
                curr_num //= 10
            if total_sum == i:
                result = i
                break
            else:
                total_sum = 0
        return result